api_tests/steps/check_oauth.py

```diff
@@ -29,4 +32,22 @@ def get_token_response(self, timeout: int = 5000, grant_type: str = 'authorizati
             data['_access_token_expiry_ms'] = timeout
 
         response = self.post(self.endpoints['token'], data=data)
-        return self.get_all_values_from_json_response(response)
\ No newline at end of file
+        return self.get_all_values_from_json_response(response)
+
+    def post(self, url: str, **kwargs) -> 'response type':
+        """Sends a post request and returns the response"""
+        try:
+            return self.session.post(url, **kwargs)
+        except requests.ConnectionError:
+            raise Exception(f"the url: {url} does not exist or is invalid")
+
+    def get_all_values_from_json_response(self, response: 'response type') -> dict:
+        """Convert json response string into a python dictionary"""
+        self._validate_response(response)
+        return json.loads(response.text)
```

Comment:
We can inline `json.loads(response.text) where it's get used and then delete these two methods. If the response is not json then it'll raise exception anyway. There is no need for validation.